### PR TITLE
Merge 4.5 into master

### PR DESCRIPTION
**Pull request by rauldpm**
|Related issue|
|---|
|Closes https://github.com/wazuh/wazuh-packages/issues/2186|

## Description

This PR merges the latest changes in 4.5 into the master branch

### CODE REVIEW

**Review by teddytpc1 — APPROVED** [review]

File: debs/SPECS/wazuh-manager/debian/rules [hunk 1]
```diff
@@ -166,11 +169,14 @@ override_dh_install:
 	cp etc/templates/config/darwin/18/sca.files ${TARGET_DIR}$(INSTALLATION_SCRIPTS_DIR)/sca/darwin/19
 	cp etc/templates/config/darwin/20/sca.files ${TARGET_DIR}$(INSTALLATION_SCRIPTS_DIR)/sca/darwin/20
 	cp etc/templates/config/darwin/21/sca.files ${TARGET_DIR}$(INSTALLATION_SCRIPTS_DIR)/sca/darwin/21
+	cp etc/templates/config/darwin/22/sca.files ${TARGET_DIR}$(INSTALLATION_SCRIPTS_DIR)/sca/darwin/22
```
Comment (verdx) [inline debs/SPECS/wazuh-manager/debian/rules:172]: File `etc/templates/config/darwin/22/sca.files` does not exist in the branch `master` of repository `wazuh/wazuh` yet

File: debs/SPECS/wazuh-manager/debian/rules [hunk 2]
```diff
@@ -128,9 +128,12 @@ override_dh_install:
 	mkdir -p ${TARGET_DIR}$(INSTALLATION_SCRIPTS_DIR)/sca/darwin/19
 	mkdir -p ${TARGET_DIR}$(INSTALLATION_SCRIPTS_DIR)/sca/darwin/20
 	mkdir -p ${TARGET_DIR}$(INSTALLATION_SCRIPTS_DIR)/sca/darwin/21
+	mkdir -p ${TARGET_DIR}$(INSTALLATION_SCRIPTS_DIR)/sca/darwin/22
```
Comment (verdx) [inline debs/SPECS/wazuh-manager/debian/rules:131]: File `etc/templates/config/darwin/22/sca.files` does not exist in the branch `master` of repository `wazuh/wazuh` yet

File: macos/package_files/build.sh [hunk 3]
```diff
@@ -60,7 +60,7 @@ function build() {
     find ${SOURCES_PATH}/src/init/ -name *.sh -type f -exec install -m 0640 {} ${INSTALLATION_SCRIPTS_DIR}/src/init \;
 
     mkdir -p ${INSTALLATION_SCRIPTS_DIR}/sca/generic
-    mkdir -p ${INSTALLATION_SCRIPTS_DIR}/sca/darwin/{15,16,17,18,20,21}
+    mkdir -p ${INSTALLATION_SCRIPTS_DIR}/sca/darwin/{15,16,17,18,20,21,22}
```
Comment (verdx) [inline macos/package_files/build.sh:63]: Same with the Darwin sca file

File: macos/package_files/build.sh [hunk 4]
```diff
@@ -72,6 +72,7 @@ function build() {
     cp ${SOURCES_PATH}/etc/templates/config/darwin/19/sca.files ${INSTALLATION_SCRIPTS_DIR}/sca/darwin/19/
     cp ${SOURCES_PATH}/etc/templates/config/darwin/20/sca.files ${INSTALLATION_SCRIPTS_DIR}/sca/darwin/20/
     cp ${SOURCES_PATH}/etc/templates/config/darwin/21/sca.files ${INSTALLATION_SCRIPTS_DIR}/sca/darwin/21/
+    cp ${SOURCES_PATH}/etc/templates/config/darwin/22/sca.files ${INSTALLATION_SCRIPTS_DIR}/sca/darwin/22/
```
Comment (verdx) [inline macos/package_files/build.sh:75]: Same

File: rpms/SPECS/wazuh-manager.spec [hunk 5]
```diff
@@ -108,8 +116,8 @@ mkdir -p ${RPM_BUILD_ROOT}%{_localstatedir}/packages_files/manager_installation_
 mkdir -p ${RPM_BUILD_ROOT}%{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/{applications,generic}
 mkdir -p ${RPM_BUILD_ROOT}%{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/amzn/{1,2}
 mkdir -p ${RPM_BUILD_ROOT}%{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/centos/{8,7,6,5}
-mkdir -p ${RPM_BUILD_ROOT}%{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/darwin/{15,16,17,18,19,20,21}
-mkdir -p ${RPM_BUILD_ROOT}%{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/debian/{7,8,9}
+mkdir -p ${RPM_BUILD_ROOT}%{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/darwin/{15,16,17,18,19,20,21,22}
```
Comment (verdx) [inline rpms/SPECS/wazuh-manager.spec:119]: Same

File: rpms/SPECS/wazuh-manager.spec [hunk 6]
```diff
@@ -757,6 +762,8 @@ rm -fr %{buildroot}
 %attr(640, root, wazuh) %config(missingok) %{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/darwin/20/*
 %dir %attr(750, wazuh, wazuh) %config(missingok) %{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/darwin/21
 %attr(640, root, wazuh) %config(missingok) %{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/darwin/21/*
+%dir %attr(750, wazuh, wazuh) %config(missingok) %{_localstatedir}/tmp/sca-%{version}-%{release}-tmp/darwin/22
```
Comment (verdx) [inline rpms/SPECS/wazuh-manager.spec:765]: Same

**Comment by verdx** [comment]
The tests are failing because the SCA files for Debian 11 and Darwin 22 have been introduced in the Debian SPECS but have not yet been added to branch `master` in the sources: https://github.com/wazuh/wazuh/tree/master/etc/templates/config/debian

<details>
  <summary>Error in test Build Wazuh Packages - DEB</summary>

```
cp etc/templates/config/debian/11/sca.files /build_wazuh/agent/wazuh-agent-4.6.0/debian/wazuh-agent"/var/ossec/packages_files/agent_installation_scripts"/sca/debian/11
cp: cannot stat `etc/templates/config/debian/11/sca.files': No such file or directory
make[1]: *** [override_dh_install] Error 1
make[1]: Leaving directory `/build_wazuh/agent/wazuh-agent-4.6.0'
make: *** [binary] Error 2
dpkg-buildpackage: error: sudo debian/rules binary gave error exit status 2
debuild: fatal error at line 1357:
dpkg-buildpackage -rsudo -d -us -uc -ai386 -b failed
Error: Process completed with exit code 1.
```
</details>

**Review by verdx — APPROVED** [review]
The errors in the tests and the previous review are expected because of the order in which the different repositories are being merged.

**Pull request merged**